source/extensions/filters/http/rate_limit_quota/filter.cc

```diff
@@ -28,27 +28,35 @@ Http::FilterHeadersStatus RateLimitQuotaFilter::decodeHeaders(Http::RequestHeade
   // succeeds.
   const RateLimitOnMatchAction& match_action =
       match_result.value()->getTyped<RateLimitOnMatchAction>();
-  auto ret = match_action.generateBucketId(*data_ptr_, factory_context_, visitor_);
+  absl::StatusOr ret = match_action.generateBucketId(*data_ptr_, factory_context_, visitor_);
   if (!ret.ok()) {
     // When it failed to generate the bucket id for this specific request, the request is ALLOWED by
     // default (i.e., fail-open).
     ENVOY_LOG(debug, "Unable to generate the bucket id: {}", ret.status().message());
     return Envoy::Http::FilterHeadersStatus::Continue;
   }
 
-  BucketId bucket_id_proto = ret.value();
+  const BucketId& bucket_id_proto = *ret;
   const size_t bucket_id = MessageUtil::hash(bucket_id_proto);
   ENVOY_LOG(trace, "Generated the associated hashed bucket id: {} for bucket id proto:\n {}",
             bucket_id, bucket_id_proto.DebugString());
+
+  ProtobufWkt::Struct bucket_log;
+  auto* bucket_log_fields = bucket_log.mutable_fields();
+  for (const auto& bucket : bucket_id_proto.bucket())
+    (*bucket_log_fields)[bucket.first] = ValueUtil::stringValue(bucket.second);
+
+  callbacks_->streamInfo().setDynamicMetadata(
+      "envoy.extensions.http_filters.rate_limit_quota.bucket", bucket_log);
```

Comment:
nit: Can you define a string const for key `"envoy.extensions.http_filters.rate_limit_quota.bucket"`?

But it is really up to you.

Reply:
Updated.